File: backend/app/db/models/scan.py
from sqlalchemy import Column, String, Float, DateTime, ForeignKey, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from datetime import datetime, timezone
import uuid
from app.db.models.base import Base


class Scan(Base):
    __tablename__ = "scans"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id = Column(UUID(as_uuid=True), nullable=True, index=True)
    input_type = Column(String(20), nullable=False)
    risk_level = Column(String(20), nullable=False)
    confidence_score = Column(Float, nullable=False)
    explanation = Column(Text, nullable=False, default="")
    created_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))

    findings = relationship("Finding", back_populates="scan", cascade="all, delete-orphan")


class Finding(Base):
    __tablename__ = "findings"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    scan_id = Column(UUID(as_uuid=True), ForeignKey("scans.id", ondelete="CASCADE"), nullable=False, index=True)
    finding_type = Column(String(100), nullable=False)
    description = Column(Text, nullable=False)
    severity = Column(String(20), nullable=False)

    scan = relationship("Scan", back_populates="findings")

File: frontend/types/scan.ts
export type RiskLevel = "low" | "medium" | "high" | "critical";

export type InputType = "url" | "image" | "text" | "qr";

export interface ScanResult {
  id: string;
  inputType: InputType;
  riskLevel: RiskLevel;
  confidenceScore: number;
  explanation: string;
  findings: Finding[];
  createdAt: string;
}

export interface Finding {
  id: string;
  findingType: string;
  description: string;
  severity: RiskLevel;
}
